engine/src/main/java/com/datatorrent/stram/debug/TupleRecorder.java

```diff
@@ -76,6 +76,33 @@
   private String recordingNameTopic;
   private long numWindows = Long.MAX_VALUE; // number of windows to record
   private Runnable stopProcedure; // stop procedure to execute
+
+  private static final Logger logger = LoggerFactory.getLogger(TupleRecorder.class);
+
+  // If there are errors processing tuples, don't log an error for every tuple as it could overwhelm the logs.
+  // The property specifies the minumum number of tuples between two consecutive error log statements. Set it to zero to
+  // log every tuple error
+  private static int ERROR_LOG_GAP;
+  int lastLog = -1;
+
+  static {
```

Comment:
There can be multiple TupleRecorder instances. Don't want to be doing special initializations the first time any instance is created and associated synchronization if multiple instances can be created at the same time.